### PR TITLE
scripts: use GOPATH or GOBIN prefix 🙏

### LINKED ISSUE #2555: update-codegen assumes deepcopy-gen on path
This is at HEAD:
```
+ ./hack/update-codegen.sh
Generating deepcopy funcs for resource:v1alpha1
/workspace/source//hack/generate-groups.sh: line 69: deepcopy-gen: command not found
```

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This will use GOPATH or GOBIN (if set) to find the installed binaries.
Closes #2555 

/kind cleanup

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

/cc @afrittoli @sbwsg @mattmoor 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
Describe any user facing changes here, or delete this block.

Examples of user facing changes:
- API changes
- Bug fixes
- Any changes in behavior

```

### CODE REVIEW

File: hack/generate-groups.sh [hunk 1]
```diff
@@ -51,6 +51,11 @@ shift 4
   go install k8s.io/code-generator/cmd/{defaulter-gen,client-gen,lister-gen,informer-gen,deepcopy-gen}
 )
 
+PREFIX="${GOPATH}"
```
Comment (mattmoor) [inline hack/generate-groups.sh:54]: shouldn't this have `/bin` at the end?

Comment (vdemeester) [reply hack/generate-groups.sh:54]: it should :facepalm:

File: hack/generate-groups.sh [hunk 2]
```diff
@@ -51,6 +51,11 @@ shift 4
   go install k8s.io/code-generator/cmd/{defaulter-gen,client-gen,lister-gen,informer-gen,deepcopy-gen}
 )
 
+PREFIX="${GOPATH}/bin"
+if [[ -n "${GOBIN}" ]]; then
+    PREFIX="${GOBIN}"
+fi
+
```
Comment (chmouel) [inline hack/generate-groups.sh:58]: You can shorten those lines with :

```
PREFIX=${GOBIN:-${PREFIX}/bin}
```

File: hack/generate-knative.sh [hunk 3]
```diff
@@ -50,6 +50,11 @@ shift 4
   go install knative.dev/pkg/codegen/cmd/injection-gen
 )
 
+PREFIX="${GOPATH}/bin"
+if [[ -n "${GOBIN}" ]]; then
+    PREFIX="${GOBIN}"
+fi
+
```
Comment (chmouel) [inline hack/generate-knative.sh:57]: Same as early

**Pull request merged**